Testnet/MintableToken/MintableToken/MintableToken.cs

```diff
@@ -211,14 +292,63 @@ public void Mint(Address account, UInt256 amount)
             PreviousSupply = (this.TotalSupply - amount),
             TotalSupply = this.TotalSupply
         });
+    }   
+
+    public void Mint(Address account, UInt256 amount)
+    {
+        Assert(Message.Sender == Interflux, "Only Interflux can call this method");
+
+        InternalMint(account, amount);
     }
 
     /// <inheritdoc />
-    public void MintWithMetadata(Address account, UInt256 amount, string metadata)
+    private void MintWithMetadata(Address account, UInt256 amount, UInt256 fee, string metadata)
     {
-        Mint(account, amount);
+        InternalMint(account, amount);
 
-        Log(new MintMetadata() { To = account, Amount = amount, Metadata = metadata });
+        Log(new MintMetadata() { To = account, Amount = amount, Metadata = metadata, Fee = fee });
+
+        Assert(TransferFrom(account, this.Owner, fee), "Fee transfer failed");
+    }
+
+    /// <inheritdoc />
+    public void MintWithMetadataForNetwork(Address account, UInt256 amount, UInt256 fee, string metadata, string destinationAccount, string destinationNetwork)
```

Comment:
It is possible that fee can be manipulated or at least mistakenly passed lower fee in here ?

Reply:
Only the contract owner can make this call.

Reply:
I think it would be good to add an explicit check for `fee >= 0`, since the `MintWithMetadata` doesn't directly check for that either. The `TransferFrom` logic should catch it, but it seems better to catch an unexpected condition early.

Reply:
Whoops, ignore that, unsigned int

Reply:
I suppose my reasoning was more along the lines of, shouldn't we also check that `fee <= amount` for safety?